### DISCUSSION TITLE
Installing build system requirements from private repositories

### DISCUSSION THREAD

**Discussion by bitfoo1** [python-poetry/poetry#8177] [Q&A]
Hi, maybe I didn't found it in the documentation.
I have to load **all** python packages from an internal proxy of pypi.org. 
I am trying to adopt `poetry` with `poetry-dynamic-versioning` plugin. 
I added `poetry-dynamic-versioning` with `poetry self add poetry-dynamic-versioning`, but it uses pypi.org to install this dependency. 
```
[[tool.poetry.source]]
name = "default-pypi-3rdparty"
url = "INTERNAL_PYPI_ORG_PROXY_URL"
priority = "default"
```
is also configured. 

Why does `poetry` not using this source URL to load the plugin?

Thank you for your help and this awesome project!

**Comment by PeterBosch2** [comment]
Same question here ... sorry, no help!

**Comment by clintonroy** [comment]
Not sure about self add to be honest, but you you want to add a python package to a virtual environment, same as any other virtual environment.  if you've installed with pipx, it has pipx inject. if you've installed in another way, the first line of poetry should hopefully point to the python in your poetry venv, use that to install a package, something like.  /home/clintonr/.local/pipx/venvs/poetry/bin/python -m pip install --index-url internal.pypi package

**Comment by PeterBosch2** [comment]
Some "black magic" to do the trick: Poetry manages its own installation using a `pyproject.toml` file in `~/.config/pypoetry`. You can use `poetry` to manipulate this `pyproject.toml`:
- make sure, that `~/.config/pypoetry/pyproject.toml` exists -- it is just created on demand!
- let `poetry` add a new default source to this `pyproject.toml`
- then issue your `poetry self add poetry-dynamic-versioning` 

```bash
set -euo pipefail

# trigger to create ~/.config/pypoetry/pyproject.toml unless it already exists
poetry self show || true

# make sure, ~/.config/pypoetry/pyproject.toml  exists
[ -r ~/.config/pypoetry/pyproject.toml ]

# add your local repository
poetry -C ~/.config/pypoetry source add my-pypi-mirror --default "${MY_PYPI_MIRROR_URL}"

# install your plugin
export POETRY_HTTP_BASIC_MY_PYPI_MIRROR_USERNAME="${your_username}"
export POETRY_HTTP_BASIC_MY_PYPI_MIRROR_PASSWORD="${your_password}"
poetry self add poetry-dynamic-versioning
```

However, I understand that this is not an officially supported solution.
